python/lsst/ip/isr/photodiode.py

```diff
@@ -0,0 +1,260 @@
+
+        The dictionary should be able to be round-tripped through.
+        `fromDict`.
+
+        Returns
+        -------
+        dictionary : `dict`
+            Dictionary of properties.
+        """
+        self.updateMetadata()
+
+        outDict = {}
+        metadata = self.getMetadata()
+        outDict['metadata'] = metadata
+
+        outDict['timeSamples'] = self.timeSamples.tolist()
+        outDict['currentSamples'] = self.currentSamples.tolist()
+
+        outDict['integrationMethod'] = self.integrationMethod
+
+        return outDict
+
+    @classmethod
+    def fromTable(cls, tableList):
+        """Construct calibration from a list of tables.
+
+        This method uses the `fromDict` method to create the
+        calibration after constructing an appropriate dictionary from
+        the input tables.
+
+        Parameters
+        ----------
+        tableList : `list` [`astropy.table.Table`]
+            List of tables to use to construct the crosstalk
+            calibration.
+
+        Returns
+        -------
+        calib : `lsst.ip.isr.PhotodiodeCalib`
+            The calibration defined in the tables.
+        """
+        dataTable = tableList[0]
+
+        metadata = dataTable.meta
+        inDict = {}
+        inDict['metadata'] = metadata
+        inDict['integrationMethod'] = metadata.pop('INTEGRATION_METHOD', None)
+
+        inDict['timeSamples'] = dataTable['TIME']
+        inDict['currentSamples'] = dataTable['CURRENT']
+
+        return cls().fromDict(inDict)
+
+    def toTable(self):
+        """Construct a list of tables containing the information in this
+        calibration.
+
+        The list of tables should create an identical calibration
+        after being passed to this class's fromTable method.
+
+        Returns
+        -------
+        tableList : `list` [`astropy.table.Table`]
+            List of tables containing the photodiode calibration
+            information.
+        """
+        self.updateMetadata()
+        catalog = Table([{'TIME': self.timeSamples,
+                          'CURRENT': self.currentSamples}])
+        inMeta = self.getMetadata().toDict()
+        outMeta = {k: v for k, v in inMeta.items() if v is not None}
+        outMeta.update({k: "" for k, v in inMeta.items() if v is None})
+        outMeta['INTEGRATION_METHOD'] = self.integrationMethod
+        catalog.meta = outMeta
+
+        return([catalog])
+
+    @classmethod
+    def readSummitPhotodiode(cls, filename):
+        """Construct a PhotodiodeCalib by reading the simple column format.
+
+        Parameters
+        ----------
+        filename : `str`
+            File to read samples from.
+
+        Returns
+        -------
+        calib : `lsst.ip.isr.PhotodiodeCalib`
+            The calibration defined in the file.
+        """
+        rawData = np.loadtxt(filename, dtype=[('time', 'float'), ('current', 'float')])
+
+        import os.path
```

Comment:
Is there a reason this is imported down here? Does this pass linting?!

Reply:
Hmm, OK, I see. I didn't think imports not at the top were legal, but if it passes, it passes!

Reply:
I'm also confused. I could have sworn that imports have to be at the start of a code block but this one comes after that variable is set.